test/lit/passes/cast-refining.wast

```diff
@@ -1011,6 +1067,106 @@
   )
 )
 
+;; As above, but now the functions are also public types (exported). We should
+;; be careful here in the future even when we do optimize function types.
+(module
+  ;; YESTNH:      (type $A (func))
+  ;; NO_TNH:      (type $A (func))
+  (type $A (func))
+
+  ;; YESTNH:      (type $funcref_=>_none (func (param funcref)))
+
+  ;; YESTNH:      (type $B (func_subtype $A))
+  ;; NO_TNH:      (type $funcref_=>_none (func (param funcref)))
+
+  ;; NO_TNH:      (type $B (func_subtype $A))
+  (type $B (func_subtype $A))
+
+  ;; YESTNH:      (type $C (func_subtype $B))
+  ;; NO_TNH:      (type $C (func_subtype $B))
+  (type $C (func_subtype $B))
+
+  ;; YESTNH:      (export "A" (func $A))
+
+  ;; YESTNH:      (export "C" (func $C))
+
+  ;; YESTNH:      (func $A (type $A)
+  ;; YESTNH-NEXT:  (nop)
+  ;; YESTNH-NEXT: )
+  ;; NO_TNH:      (export "A" (func $A))
+
+  ;; NO_TNH:      (export "C" (func $C))
+
+  ;; NO_TNH:      (func $A (type $A)
+  ;; NO_TNH-NEXT:  (nop)
+  ;; NO_TNH-NEXT: )
+  (func $A (export "A") (type $A)
+  )
+
+  ;; YESTNH:      (func $C (type $A)
+  ;; YESTNH-NEXT:  (nop)
+  ;; YESTNH-NEXT: )
+  ;; NO_TNH:      (func $C (type $A)
+  ;; NO_TNH-NEXT:  (nop)
+  ;; NO_TNH-NEXT: )
+  (func $C (export "C") (type $A)
+  )
```

Comment:
Is this intended to have `(type $A)`? It looks like it's meant to have `(type $C)`, but that would break closed-world validation by making `$B` public, so maybe this should be removed.

Reply:
Right, good point, this should be type `$C` but not exported.